pkg/logs/input/file/position.go

```diff
@@ -12,25 +12,82 @@ import (
 	"github.com/DataDog/datadog-agent/pkg/logs/auditor"
 )
 
+// TailingMode type
+type TailingMode uint8
+
+// Tailing Modes
+const (
+	ForceBeginning = iota
+	ForceEnd
+	Beginning
+	End
+)
+
+// Tailing mode string representation
+const (
+	ForceBeginningStr = "forceBeginning"
+	ForceEndStr       = "forceEnd"
+	BeginningStr      = "beginning"
+	EndStr            = "end"
+)
+
+var tailingModeToStringRepresentations = map[TailingMode]string{
+	ForceBeginning: ForceBeginningStr,
+	ForceEnd:       ForceEndStr,
+	Beginning:      BeginningStr,
+	End:            EndStr,
+}
+
+// TailingModeFromString parses a string and returns a corresponding tailing mode, default to End if not found
+func TailingModeFromString(mode string) (TailingMode, bool) {
+	for m, mStr := range tailingModeToStringRepresentations {
```

Comment:
We could have a map representation -> mode, also the gain with 4 entries is very small, we could just have an array of pairs and reuse it for both `TailingModeFromString` and `String` ?

Reply:
I guess with only 4 entries we can go with an array.